src/rgw/rgw_http_client.h

```diff
@@ -178,36 +178,45 @@ class RGWHTTPHeadersCollector : public RGWHTTPClient {
 };
 
 
-class RGWPostHTTPData : public RGWHTTPClient {
-  bufferlist *bl;
+class RGWHTTPTransceiver : public RGWHTTPHeadersCollector {
```

Comment:
Yes, I know this naming is awkward. Unfortunately, I cannot find anything better. :-(

Guys, do have an idea? This class basically sends a request using an HTTP method specified by caller (so, `RGWPostHTTPData` doesn't fit IMO) and collects response.